docs/HANDLED_RULES.md

```diff
@@ -141,6 +141,42 @@ Example:
 
 -----
 
+#### JEE applications should not "getClassLoader" ([Sonar Rule 3032](https://rules.sonarsource.com/java/RSPEC-3032))
+
+Using the standard getClassLoader() may not return the right class loader in Java Enterprise Edition context, i.e. the code use any of these [javax packages](https://docs.oracle.com/javaee/6/api/overview-summary.html). Instead, go through the currentThread. In particular, this processor will transform the code in `.java` files if they appear to use the `javax` package.
+
+Example:
+```diff
++    ClassLoader d = Thread.currentThread().getContextClassLoader();
+-    ClassLoader d = this.getClass().getClassLoader(); // Noncompliant
++    public void case1() {
++        ClassLoader c = Thread.currentThread().getContextClassLoader();
++
++    }
+-    public void case1() {
+-        ClassLoader c = this.getClass().getClassLoader(); // Noncompliant
+-    }
++    public void case2() throws ClassNotFoundException {
++        Thread.currentThread().getContextClassLoader().loadClass("anotherclass");
++    }
+-    public void case2() throws ClassNotFoundException{
+-        Dummy.class.getClassLoader().loadClass("anotherclass"); // Noncompliant
+-    }
++    abstract class InnerClass {
++        ClassLoader f = Thread.currentThread().getContextClassLoader();
++        public void hello() {
++            Thread.currentThread().getContextClassLoader();
++        }
++    }
+-    abstract class InnerClass {
+-        ClassLoader f = this.getClass().getClassLoader(); // Noncompliant 
+-        public void hello() {
+-            this.getClass().getClassLoader(); // Noncompliant
+-       }
+-    }
```

Comment:
Too many examples. Here, it's just to give one or two examples of transformations. Select the best cases. Here, I think that the best are:

-     ClassLoader d = this.getClass().getClassLoader(); // Noncompliant
+    ClassLoader d = Thread.currentThread().getContextClassLoader();

-     Dummy.class.getClassLoader().loadClass("anotherclass"); // Noncompliant
+    Thread.currentThread().getContextClassLoader().loadClass("anotherclass");

Also, at least here in the documentation, always show the deleted line and then the added line, and not the other way around as you did. The deleted line followed by the added line represents better the "before and after".